### PR TITLE
[karma-eas-attestation] EAS Attestation Proposal Validation 

### PR DESCRIPTION
**Pull request by andremury**
> Replace for #1233

Changes proposed in this pull request:
- [X] New validation schema using EAS's Attestation. 
- [X] Removed line checking strategies (strategies are not used n this validation)
- [X] Included support for `Optimism` and `Arbitrum`

Use [Attest.sh](http://attest.sh) to allow users to create a new proposal in the DAO's space

#### Tests: passing
image

### CODE REVIEW

File: src/validations/eas-attestation/index.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+export default class extends Validation {
+  public id = 'eas-attestation';
+  public github = 'karmahq';
+  public version = '0.1.0';
+  public title = 'EAS Attestation';
+  public description =
+    'Use EAS attest.sh to determine if user can create a proposal.';
+
```
Comment (ChaituVR) [inline src/validations/eas-attestation/index.ts:75]:
```suggestion
  public description =
    'Use EAS attest.sh to determine if user can create a proposal.';
  public proposalValidationOnly = true;
```

File: src/validations/eas-attestation/index.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  const index = response.data.schema.attestations.findIndex(
+    (a) => a.recipient.toLowerCase() === address.toLowerCase() && !a.revoked
+  );
+
+  return index >= 0;
+}
+
+export default class extends Validation {
+  public id = 'eas-attestation';
+  public github = 'karmahq';
```
Comment (ChaituVR) [inline src/validations/eas-attestation/index.ts:70]:
```suggestion
  public github = 'show-karma';
```

File: src/validations/eas-attestation/examples.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+[
+  {
+    "name": "Example of a EAS Attestation",
+    "author": "0xd7d1DB401EA825b0325141Cd5e6cd7C2d01825f2",
+    "space": "0xmury.eth",
+    "network": "11155111",
+    "snapshot": "latest",
+    "params": {
+      "schemaId": "0xc0f979976278d9e1d4fa97b7270c0cc07835aa5f27dd897a871b3332ec6cff22"
+    }
```
Comment (ChaituVR) [inline src/validations/eas-attestation/examples.json:10]: Could you add a schema.json file with the below:

```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$ref": "#/definitions/Validation",
  "definitions": {
    "Validation": {
      "title": "Eas attestation",
      "type": "object",
      "properties": {
        "schemaId": {
          "type": "string",
          "title": "Schema Id",
          "examples": ["e.g. 0xc0f979976278d9e1d4fa97b7270c0cc07835aa5f27dd897a871b3332ec6cff22"]
        }
      },
      "required": [],
      "additionalProperties": false
    }
  }
}

```

Comment (andremury) [reply src/validations/eas-attestation/examples.json:10]: added

File: src/validations/karma-eas-attestation/index.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import Validation from '../validation';
+import fetch from 'cross-fetch';
+
+export const author = 'show-karma';
+export const version = '0.1.1';
+
```
Comment (ChaituVR) [inline src/validations/karma-eas-attestation/index.ts:6]:
```suggestion
```

File: src/validations/karma-eas-attestation/index.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+export default class extends Validation {
+  public id = 'karma-eas-attestation';
+  public github = 'karmahq';
+  public version = '0.1.0';
+  public title = 'Karma EAS Attestation';
+  public description =
+    'Use EAS attest.sh to determine if user can create a proposal.';
+
```
Comment (ChaituVR) [inline src/validations/karma-eas-attestation/index.ts:75]:
```suggestion
  public description =
    'Use EAS attest.sh to determine if user can create a proposal.';
  public proposalValidationOnly = true;
```

**Review by ChaituVR — CHANGES_REQUESTED** [review]

**Comment by maheshmurthy** [comment]
@ChaituVR I think this is ready to go now? Can you please approve so we can start using it? Thank you!

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**